app/controllers/favorites_controller.rb

```diff
@@ -10,4 +14,18 @@ def update
     flash[:add_favorite] = "This pet was added to My Favorites. You now have #{pluralize(quantity, "favorite")}"
     redirect_to "/pets/#{pet.id}"
   end
+
+  def destroy
+    pet = Pet.find(params[:pet_id])
+    favorites.remove_pet(pet.id)
+    session[:favorite] = favorites.contents
+    quantity = favorites.count
+    flash[:remove_favorite] = "This pet was removed from My Favorites. You now have #{pluralize(quantity, "favorite")}"
+    redirect_to request.referer
```

Comment:
This one is awesome!!!! Good Job!!! Thank you for explaining this to me.

Reply:
For future research: https://api.rubyonrails.org/classes/ActionController/Redirecting.html